src/main/java/com/blibli/future/service/EmployeeTappingServiceImpl.java

```diff
@@ -167,22 +171,20 @@ public boolean addTapData(List<TapData> tapDataList) {
         }
     }
 
+    @Override
     public boolean addTapMachineFile(MultipartFile tapMachineFile) {
         List<TapData> tapDatas = new ArrayList<>();
-        List<String> inputData;
 
-        String fileName = tapMachineFile.getOriginalFilename();
+        List<String> inputData = fileReaderComponent.readFileAsStrings(tapMachineFile);
 
-        if(fileName.endsWith(".csv")) {
-            inputData = readCsvAsList(tapMachineFile);
-        } else {
-            return false; //Or should it throws some "Unsupported file Type" ?
+        if(inputData == null) {
```

Comment:
If you throw exception while reading the file (in below) you became forced to handle it. No surprise result when you *forget* to handle special case / broken input.